Add Roman to Integer solution and tests

DieHard073055 opened the pull request:
## Implement Roman to Integer Solution

This PR adds a `Solution` struct and implementation for the Roman to Integer problem.

### Solution overview:
The solution involves iterating through the input Roman numeral string and converting each character to its corresponding integer value using the `roman_char_to_int` function. During the iteration, the `eval_roman_chars` function is used to compare the current Roman character with the next one. This helps to identify and handle the cases where subtraction is required, such as 'IV' and 'IX'. If subtraction is needed, the function returns a tuple containing a boolean value `true` (indicating to skip the next character) and the integer value resulting from the subtraction. Otherwise, the function returns `false` and the integer value of the current Roman character.

The `roman_to_int` function iterates through the input string, applying the `eval_roman_chars` function to determine whether to add or subtract the corresponding integer values. It keeps track of the total value in the `total` variable. Once the iteration is complete, the `total` value is returned as the integer representation of the input Roman numeral.

### Implementation details:
- A `roman_char_to_int` function that converts a single Roman character to its corresponding integer value.
- An `eval_roman_chars` function that evaluates two Roman characters and returns a tuple containing a boolean value to indicate whether to skip the next character and an integer value to be added to the total.
- A `roman_to_int` function that converts a given Roman numeral string to its integer value.

Additionally, tests have been added to verify the correctness of the solution.

DieHard073055 committed:
Add Roman to Integer solution and tests

DieHard073055 committed:
Add README.md for solution overview

pamdayne commented on roman-to-integer/src/lib.rs:
```diff
@@ -0,0 +1,84 @@
+            'C' => 100,
+            'D' => 500,
+            'M' => 1000,
+            _ => !unreachable!("There should'nt be any more roman characters"),
```
typo `shouldn't`

DieHard073055 replied:
Done

pamdayne commented on roman-to-integer/src/lib.rs:
```diff
@@ -0,0 +1,84 @@
+        }
+    }
+    fn eval_roman_chars(next: i32, previous: i32) -> (bool, i32) {
+        // println!("eval roman char: next {:}, previous {:}", next, previous);
```
should we clean up these `println`'s? 
im guessing these are for debugging purposes.

DieHard073055 replied:
Done

DieHard073055 committed:
Add MyQueue implementation using VecDeque

This commit introduces the implementation of the MyQueue struct using VecDeque from the standard library. The MyQueue struct provides the following methods:

- `new`: Initializes an empty queue.
- `push`: Adds an element to the front of the queue.
- `pop`: Removes and returns the element from the back of the queue.
- `peek`: Returns the element at the back of the queue without removing it.
- `empty`: Checks if the queue is empty.

Additionally, tests have been added to verify the correctness of the MyQueue implementation.

DieHard073055 committed:
chore: remove all printlns

DieHard073055 committed:
chore: fix typo

DieHard073055 committed:
chore: clean up

DieHard073055 committed:
chore: clean up
